In [1]:
import pandas as pd

df_A = pd.read_csv("matrixA_named.csv")
df_B = pd.read_csv("matrixB_named.csv")

In [2]:
df_A

,Unnamed: 0,X [m],Y [m],Z [m],Vx [m/s],Vy [m/s],Vz [m/s],Speed [m/s],Opening Angle [°],Roll Angle Left [°],Roll Angle Right [°],Yaw Angle Left [°],Yaw Angle Right [°],Stalling Angle Left [°],Stalling Angle Right [°]
0,X [m],0.999108,0.003280,-0.001238,0.007328,-0.010696,-0.001597,-0.001238,0.000238,-0.000224,-0.000111,0.000054,0.000075,0.000415,0.000127
1,Y [m],-0.000021,0.999430,-0.000029,0.000056,0.051721,-0.000029,-0.000030,0.000006,-0.000028,-0.000024,-0.000014,0.000008,0.000040,-0.000039
2,Z [m],-0.000418,-0.001593,0.999429,-0.000024,0.004927,0.004888,-0.000098,-0.000091,0.000022,0.000039,0.000012,-0.000061,0.000026,0.000020
3,Vx [m/s],0.009650,0.017490,0.014399,0.932005,-0.068273,0.036245,0.036198,0.011567,0.005362,-0.005529,-0.006525,0.008849,-0.011379,-0.010606
4,Vy [m/s],0.000358,0.069636,0.000087,-0.001162,0.756712,0.001148,0.000667,0.002077,-0.002742,-0.002977,0.001730,0.003904,0.000715,-0.001847
5,Vz [m/s],-0.024896,0.001741,-0.049555,0.001027,-0.031696,0.976963,-0.008200,-0.005033,-0.002482,-0.001039,0.003362,-0.000960,0.005478,-0.000813
6,Speed [m/s],0.024158,0.013197,0.043358,-0.034381,-0.040392,0.023231,0.519232,0.011558,0.005777,-0.004241,-0.007561,0.007937,-0.014156,-0.010123
7,Opening Angle [°],0.075184,-0.009159,0.122506,-0.247649,-0.007277,0.131399,0.137594,0.482370,-0.001936,-0.001840,0.014808,-0.006512,0.006904,0.009481
8,Roll Angle Left [°],0.006611,-0.048344,0.029366,-0.084815,0.287226,0.040911,0.048239,-0.001213,0.489095,-0.003925,0.000705,-0.008302,0.041765,-0.011331
9,Roll Angle Right [°],-0.021923,-0.037827,-0.051379,0.079756,0.245500,-0.038493,-0.044920,0.008603,-0.002139,0.491101,-0.018937,0.002334,0.001608,-0.034522


V matriki A lahko opazimo:
- hitrost v navpični smeri in celotna hitrost na sam skok ne vplivata. sta popolnoma neodvisni od ostalih meritev.
- po diagonali imamo:
    - približno 1 za x, y, z, vx -> vy je suspiciously low
    - 0 (kot vse ostalo) za vz, speed
    - približno 0.5 za vse kote

Problemi: 
- kako je Vz neodvisen od Vz...it should be at least 1. enako za speed
- vx od x bi moral biti okoli 0.5

Gpt opinion:
- you should be worried, but not about the model, the data itself seems faulty.

In [3]:
df_B

,Unnamed: 0,takeoff_Speed,mid_Speed,landing_Speed,takeoff_Tangent,mid_Tangent,landing_Tangent,takeoff_Cross,mid_Cross,landing_Cross,...,mid_Turbulence_quad,landing_Turbulence_quad,Speed [m/s],Opening Angle [°],Roll Angle Left [°],Roll Angle Right [°],Yaw Angle Left [°],Yaw Angle Right [°],Stalling Angle Left [°],Stalling Angle Right [°]
0,X [m],0.004455,-0.001283,-0.007981,0.004134,-0.000851,-0.001009,-0.010274,0.004693,0.005206,...,0.080638,0.001967,-0.001473,0.000232,-0.000227,-0.000119,0.000048,0.000089,0.000417,0.000122
1,Y [m],0.000776,-0.000136,-0.000374,-0.000212,-0.000023,0.000059,0.000350,0.000008,-0.000131,...,-0.000477,-0.000246,-0.000030,0.000006,-0.000027,-0.000024,-0.000014,0.000008,0.000040,-0.000039
2,Z [m],0.002023,0.003608,0.001547,-0.000135,0.000280,0.000499,0.003482,-0.003625,0.001973,...,-0.023705,-0.009267,0.000084,-0.000102,0.000022,0.000041,0.000019,-0.000062,0.000023,0.000022
3,Vx [m/s],-0.087882,-0.058958,-0.102910,0.045977,-0.024318,-0.040168,-0.015334,0.089428,0.057630,...,0.272641,-0.229540,0.036782,0.011642,0.005350,-0.005484,-0.006557,0.008844,-0.011382,-0.010598
4,Vy [m/s],0.004080,0.049175,-0.043645,0.005798,-0.006553,-0.019781,-0.129994,-0.071905,0.032924,...,-0.027227,0.062871,0.000668,0.002077,-0.002742,-0.002977,0.001730,0.003904,0.000715,-0.001847
5,Vz [m/s],-0.055751,0.101287,0.084275,-0.047535,-0.014437,0.085768,0.099324,-0.184074,-0.062531,...,-0.593811,-0.413170,-0.008108,-0.005079,-0.002474,-0.001043,0.003349,-0.000942,0.005478,-0.000811
6,Speed [m/s],0.024376,-0.014360,-0.171324,0.048345,-0.015583,-0.081187,-0.057008,0.112002,0.080575,...,0.398832,0.068820,0.520352,0.011533,0.005841,-0.004187,-0.007555,0.007942,-0.014153,-0.010125
7,Opening Angle [°],-0.521815,-0.154968,0.467637,0.044069,-0.057454,0.069457,-0.047284,-0.040028,-0.193069,...,-0.400813,0.709431,0.137656,0.482376,-0.001933,-0.001850,0.014812,-0.006502,0.006904,0.009481
8,Roll Angle Left [°],-0.442270,0.053894,0.120383,0.084422,-0.062590,0.111018,0.488824,-0.143159,-0.419600,...,0.386928,-0.930745,0.048310,-0.001209,0.489095,-0.003926,0.000702,-0.008307,0.041765,-0.011331
9,Roll Angle Right [°],0.713381,0.092246,-0.637010,-0.164037,0.037869,-0.044171,-0.259396,0.153359,0.260939,...,-0.068539,-0.069561,-0.044914,0.008612,-0.002143,0.491104,-0.018937,0.002333,0.001609,-0.034522


- vx, vy in koti so vsi precej odvisni od turbulence
- opening angle je močno odvisen od turbulence in faktorjev ob pristajanju
- koti po diagonali imajo sami s sabo podobno 0.5 kot pti A matriki
- hitrost ima same ničle
- landing turbulenca je zelo odvisna od vseh kotov
- korelacija x, y, z je zelo enakomerna pri features
    -turbulenca ima največji vpliv na vse, zelo visok vpliv na x in dokaj visok vpliv na z koordinato, vendar nekoliko nižji na y

problemi so podobni kot prej

gpt opinion ni bil nič pametnega

## changed matrices

Previous findings niso več čist res

In [5]:
correlation_matrix_A = df_A.iloc[:, 1:].corr()
correlation_matrix_B = df_B.iloc[:, 1:].corr()

strong_correlations_A = correlation_matrix_A.unstack().sort_values(ascending=False)
strong_correlations_B = correlation_matrix_B.unstack().sort_values(ascending=False)

strong_correlations_A = strong_correlations_A[strong_correlations_A < 1]
strong_correlations_B = strong_correlations_B[strong_correlations_B < 1]

strong_correlations_A

Roll Angle Left [°]      Vy [m/s]                    0.257066
Vy [m/s]                 Roll Angle Left [°]         0.257066
                         Roll Angle Right [°]        0.209756
Roll Angle Right [°]     Vy [m/s]                    0.209756
Speed [m/s]              Opening Angle [°]           0.157306
                                                       ...   
Vx [m/s]                 Stalling Angle Right [°]   -0.244583
                         Stalling Angle Left [°]    -0.252112
Stalling Angle Left [°]  Vx [m/s]                   -0.252112
Speed [m/s]              Yaw Angle Right [°]        -0.259428
Yaw Angle Right [°]      Speed [m/s]                -0.259428
Length: 182, dtype: float64

In [6]:
strong_correlations_B

landing_Tangent_quad  landing_Cross_quad      0.940006
landing_Cross_quad    landing_Tangent_quad    0.940006
landing_Tangent       landing_Tangent_quad    0.902676
landing_Tangent_quad  landing_Tangent         0.902676
landing_Tangent       landing_Cross_quad      0.886084
                                                ...   
takeoff_Speed_quad    mid_Speed              -0.909912
landing_Tangent_quad  landing_Speed_quad     -0.936523
landing_Speed_quad    landing_Tangent_quad   -0.936523
mid_Turbulence_quad   mid_Turbulence         -0.943586
mid_Turbulence        mid_Turbulence_quad    -0.943586
Length: 992, dtype: float64

## Here are some notable correlations from the two matrices:

### Matrix A (Physical Features - Ski Jump Simulation):

Vy [m/s] and Roll Angle Left [°]: 0.257 – This suggests a moderate relationship between the vertical velocity component and the roll angle to the left.

Vy [m/s] and Roll Angle Right [°]: 0.210 – The vertical velocity also correlates moderately with the roll angle to the right.

Opening Angle [°] and Speed [m/s]: 0.157 – There's a mild positive correlation between the opening angle of the ski jump and the speed.

Vx [m/s] and Yaw Angle Right [°]: 0.127 – The horizontal velocity (Vx) seems somewhat related to the yaw angle.

Stalling Angle Right [°] and Speed [m/s]: 0.098 – A weak positive correlation between the stalling angle and speed.

### Matrix B (Trajectory Metrics - Ski Jump):

landing_Cross_quad and landing_Tangent_quad: 0.94 – Extremely strong correlation, indicating that landing cross and tangent angles are highly dependent on each other.

landing_Tangent and landing_Cross_quad: 0.89 – A high correlation between landing tangent and the cross component.

mid_Tangent and takeoff_Speed: 0.87 – Strong correlation, suggesting a link between the tangent angle at the middle of the jump and the takeoff speed.

takeoff_Tangent and mid_Cross_quad: 0.84 – A high correlation between the takeoff tangent and the middle cross component.

### Physical Insights:

In Matrix A, the correlations between the vertical velocity components (Vy) and roll angles (both left and right) suggest that the skier's body orientation (roll) might influence or be influenced by the vertical velocity, which is key in understanding the flight dynamics and the stability of the jump.

Matrix B indicates that the landing components (cross and tangent) have very high correlations, which makes sense because these two components are critical in determining the landing position and angle of the ski jump.